test-cypress/integration/sparql/sparql.menu.spec.js

```diff
@@ -1,807 +1,1034 @@
+            executeQuery();
+            getUpdateMessage().should('not.be.visible');
+            getResultsMessage()
+                .should('contain', 'Showing results from 1 to 74 of 74')
+                .and('contain', 'Query took');
+
+            verifyResultsPageLength(74);
+        });
 
-    it('Test execute query including inferred with ruleset "OWL-Horst (Optimized)"', function () {
-        let testRepo = 'test-repo' + Date.now();
-        // First navigate to create repository for this test
-        cy.navigateToPageS('Setup', 'Repositories');
-        // Create new repository
-        cy.createNewRepoS(testRepo, 'OWL-Horst (Optimized)').wait(1000);
-        cy.setRepoDefaultS(testRepo);
-        // Select the newly created repository
-        cy.selectRepoS(testRepo).wait(1000);
-
-        // Upload the wines ontology
-        cy.navigateToPageS('Import', 'RDF');
-        cy.openImportURLDialogS('https://www.w3.org/TR/owl-guide/wine.rdf');
-        cy.clickImportUrlBtn();
-        cy.clickImportBtnOnPopUpMenu().wait(1000);
-
-        let defaultQueryWithoutLimit = 'select * where { \n' +
-            '\t?s ?p ?o .\n' +
-            '}';
+        it('Test saved query link', () => {
+            const queryName = 'Add statements';
+            getSavedQueriesPopupBtn().click();
+            clickSavedQueryLinkFromPopup(queryName);
+
+            const expectedUrl = Cypress.config().baseUrl + '/sparql?savedQueryName=' + encodeURI(queryName);
+            getModal()
+                .should('be.visible')
+                .find('#clipboardURI')
+                .should('have.value', expectedUrl);
+
+            // Visit performs full page load
+            cy.visit(expectedUrl);
+            waitUntilSparqlPageIsLoaded();
+
+            getTabs().should('have.length', 2);
+            getActiveTabLink().should('have.text', queryName);
 
-        // Navigate back to SPARQL page
-        cy.navigateToPageS('SPARQL')
+            // Wait until editor is initialized with the query and then assert the whole query
+            getQueryArea().should('contain', 'INSERT DATA');
+            cy.fixture('queries/add-statement.txt').then((query) => {
+                verifyQueryAreaContains(query);
+            });
+        });
 
-        // Go to SPARQL and execute a SELECT query without a limit
-        cy.executeCustomQuery(defaultQueryWithoutLimit);
+        it('Test URL to current query', () => {
+            const query = 'SELECT ?s ?p ?o WHERE {?s ?p ?o .} LIMIT 100';
+            typeQuery(query);
 
-        // Verify that there are 7,065 results
-        cy.verifyQueryResultsS('1,000 of 7,065');
+            // Press the link icon to generate a link for a query
+            getQueryLinkBtn().click();
 
-        // Disable the inference from the ">>" icon on the right of the SPARQL editor.
-        cy.get('#inference').as('disableInference');
+            // TODO: Test with tab name -> should open such tab if it was deleted
```

Comment:
Maybe we should directly create tasks for these?!

Reply:
Or at least add them in the acceptance sheet.